pyccel/codegen/printing/fcode.py

```diff
@@ -306,133 +298,60 @@ def _print_Module(self, expr):
         body    = self._print(expr.body)
 
-        # ... TODO add other elements
-        private_funcs = [f.name for f in expr.funcs if f.is_private]
-        private = private_funcs
-        if private:
-            private = ','.join(self._print(i) for i in private)
-            private = 'private :: {}'.format(private)
-        else:
-            private = ''
-        # ...
-
-        decs    = expr.declarations
         vars_to_print = self.parser.get_variables(self._namespace)
         for v in vars_to_print:
             if v not in expr.variables:
                 decs.append(Declare(v.dtype,v))
-        func_in_func = False
-        for func in expr.funcs:
-            for i in func.body.body:
-                if isinstance(i, FunctionDef):
-                    func_in_func = True
-                    break
-        if expr.classes or expr.interfaces or func_in_func:
-            # TODO shall we use expr.variables? or have a more involved algo
-            #      we will need to walk through the expression and see what are
-            #      the variables that are needed in the definitions of classes
-            variables = []
-            for i in expr.interfaces:
-                variables += i.functions[0].global_vars
-            for i in expr.funcs:
-                variables += i.global_vars
-            variables =list(set(variables))
-            for i,d in enumerate(decs):
-                #remove variables that are declared in the modules
-                if d.variable in variables:
-                    decs[i] = None
-            decs = [i for i in decs if i]
-
-            module_utils = Module(expr.name, list(variables),
-                                  expr.funcs, expr.interfaces, expr.classes,
-                                  imports=expr.imports)
-            modules = self._print(module_utils)
-
-            imports = ('{imports}\n'
-                       'use mod_{name}\n').format(imports=imports, name=expr.name)
 
-        else:
-            # ... uncomment this later and remove it from the top
-#            decs    = '\n'.join(self._print(i) for i in expr.declarations)
-            # ...
-
-            # ...
-            sep = self._print(SeparatorComment(40))
-            funcs = ''
-            if expr.funcs:
-                for i in expr.funcs:
-                    funcs = ('{funcs}\n'
-                             '{sep}\n'
-                             '{f}\n'
-                             '{sep}\n').format(funcs=funcs, sep=sep, f=self._print(i))
-
-                funcs = 'contains\n{0}'.format(funcs)
-            # ...
+        mpi = False
+        #we use this to detect of we are using so that we can add
+        # mpi_init and mpi_finalize in the code instruction
+        # TODO should we find a better way to do this?
+        for i in expr.imports:
+            if 'mpi4py' == str(getattr(i.source,'name',i.source)):
+                mpi = True
```

Comment:
This can be written as a list comprehension
```
mpi = any( 'mpi4py' == str(getattr(i.source,'name',i.source)) for i in expr.imports )
```

Reply:
Good suggestion! Thanks